// !DIAGNOSTICS: -UNUSED_PARAMETER -PLATFORM_CLASS_MAPPED_TO_KOTLIN

<!CONFLICTING_PLATFORM_DECLARATIONS!>fun foo(s: java.lang.String)<!> {}
<!CONFLICTING_PLATFORM_DECLARATIONS!>fun foo(s: String)<!> {}